### PR TITLE
Allow Python API to be used without h5py

### PR DESCRIPTION
**Pull request by smharper**
Importing openmc in turn imports h5py through openmc.summary.  This PR makes it so that h5py isn't imported unless a user actually inits a `Summary`

### CODE REVIEW

File: openmc/summary.py [hunk 1]
```diff
@@ -2,17 +2,22 @@
 
 import openmc
 
-try:
-    import h5py
-except ImportError:
-    msg = 'Unable to import h5py which is needed by openmc.summary'
-    raise ImportError(msg)
+
+_h5_imported = False
```
Comment (wbinventor) [inline openmc/summary.py]: Why is this needed? It would be cleaner to just re-import ``h5py`` each time a ``Summary`` is created (typically once), unless there is good reason to only import once.

**Comment by wbinventor** [comment]
Good PR @smharper. I think this is fine pending my question about re-imports.

File: openmc/summary.py (hunk 1 above)
Comment (paulromano) [inline openmc/summary.py]: I agree with @wbinventor. On subsequent re-imports, if `h5py` is already in the namespace it is essentially a no-op. Also, the error message here doesn't really add much additional information beyond the normal 'No module named h5py' ImportError along with the traceback showing it was called from openmc.summary.
